guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,872 @@
+eval_ds = eval_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+
+### Optimizer
+
+In this guide, we use a standard SGD optimizer and rely on the
+[`keras.callbacks.ReduceLROnPlateau`](https://keras.io/api/callbacks/reduce_lr_on_plateau/)
+callback to reduce the learning rate.
+
+You will always want to include a `global_clipnorm` when training object
+detection models.  This is to remedy exploding gradient problems that frequently
+occur when training object detection models.
+"""
+
+base_lr = 0.005
+# including a global_clipnorm is extremely important in object detection tasks
+optimizer = tf.keras.optimizers.SGD(
+    learning_rate=base_lr, momentum=0.9, global_clipnorm=10.0
+)
+
+"""
+To achieve the best results on your dataset, you'll likely want to hand craft a
+`PiecewiseConstantDecay` learning rate schedule.
+While `PiecewiseConstantDecay` schedules tend to perform better, they don't
+translate between problems.
+"""
+
+"""
+### Loss functions
+
+You may not be familiar with the `"focal"` or `"smoothl1"` losses.  While not
+common in other models, these losses are more or less staples in the object
+detection world.
+
+In short, ["Focal Loss"](https://arxiv.org/abs/1708.02002) places extra emphasis
+on difficult training examples.  This is useful when training the classification
+loss, as the majority of the losses are assigned to the background class.
+
+"SmoothL1 Loss" is used to [prevent exploding gradients](https://arxiv.org/abs/1504.08083)
+that often occur when attempting to perform the box regression task.
+
+In KerasCV you can use these losses simply by passing the strings `"focal"` and
+`"smoothl1"` to `compile()`:
+"""
+
+pretrained_model.compile(
+    classification_loss="focal",
+    box_loss="smoothl1",
+)
+
+"""
+### Metric evaluation
+
+Just like any other metric, you can pass the `KerasCV` object detection metrics
+to `compile()`.  The most popular Object Detection metrics are COCO metrics,
+which were published alongside the MSCOCO dataset.  KerasCV provides an easy
+to use suite of COCO metrics. under the `keras_cv.metrics.BoxCOCOMetrics`
+symbol:
+"""
+
+coco_metrics = keras_cv.metrics.BoxCOCOMetrics(
+    bounding_box_format="xywh", evaluate_freq=20
+)
+
+"""
+Let's define a quick helper to print our metrics in a nice table:
+"""
+
+
+def print_metrics(metrics):
+    maxlen = max([len(key) for key in result.keys()])
+    print("Metrics:")
+    print("-" * (maxlen + 1))
+    for k, v in metrics.items():
+        print(f"{k.ljust(maxlen+1)}: {v.numpy():0.2f}")
+
+
+"""
+Due to the high computational cost of computing COCO metrics, the KerasCV
+`BoxCOCOMetrics` component requires an `evaluate_freq` parameter to be passed to
+its constructor.  Every `evaluate_freq`-th call to `update_state()`, the metric
+will recompute the result.  In between invocations, a cached version of the
```

Comment:
"cached version": essentially, the metric is returning a wrong value most of the time. This is really not the ideal solution. Please implement a proper "eval-every-N-epochs" functionality instead.